WordPress/Classes/Utility/CoreDataHelper.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+
+
+
+// MARK: - ManagedObject Base Class
+//
+protocol ManagedObject
+{
+    static var entityName: String { get }
+}
+
+
+// MARK: - Core Data Helpers
+//
+struct CoreDataHelper<T where T: NSManagedObject, T: ManagedObject>
+{
+    /// CoreData ManagedObjectContext
+    ///
+    let context: NSManagedObjectContext
+
+
+    /// Returns a new FetchRequest for the associated Entity
+    ///
+    func newFetchRequest() -> NSFetchRequest {
+        return NSFetchRequest(entityName: T.entityName)
+    }
+
+
+    /// Returns all of the entities that match with a given predicate.
+    ///
+    /// - Parameter predicate: Defines the conditions that any given object should meet. Optional.
+    ///
+    func allObjects(matchingPredicate predicate: NSPredicate? = nil, sortedBy descriptors: [NSSortDescriptor]? = nil)  -> [T] {
+        let request = newFetchRequest()
+        request.predicate = predicate
+        request.sortDescriptors = descriptors
+
+        return loadObjects(withFetchRequest: request)
+    }
+
+    /// Returns the number of entities found that match with a given predicate.
+    ///
+    /// - Parameter predicate: Defines the conditions that any given object should meet. Optional.
+    ///
+    func countObjects(matchingPredicate predicate: NSPredicate? = nil) -> Int {
+        let request = newFetchRequest()
+        request.predicate = predicate
+        request.includesSubentities = false
+        request.predicate = predicate
+
+        var result = 0
+
+        do {
+            result = try context.countForFetchRequest(request)
+        } catch {
+            DDLogSwift.logError("Error counting objects [\(T.entityName)]")
```

Comment:
Logging errors like this is not a good idea, because we're basically allowing them to silently fail during development and testing.

There are two possible scenarios as I see:
1. Either an error is not supposed to happen in release, and this class should at least throw an assertion failure (or a fatal error).
2. Or the error is supposed to happen in release, and the parent VC should handle it.

In my humble opinion these are errors of the type 1.